cli/src/pcluster/models/cluster.py

```diff
@@ -254,21 +254,33 @@ def config_presigned_url(self) -> str:
     @property
     def compute_fleet_status(self) -> ComputeFleetStatus:
         """Status of the cluster compute fleet."""
-        compute_fleet_status_manager = ComputeFleetStatusManager(self.name)
-        status = compute_fleet_status_manager.get_status()
-        if status == ComputeFleetStatus.UNKNOWN:
-            stack_status_to_fleet_status = {
-                "CREATE_IN_PROGRESS": ComputeFleetStatus.STARTING,
-                "DELETE_IN_PROGRESS": ComputeFleetStatus.STOPPING,
-                "CREATE_FAILED": ComputeFleetStatus.STOPPED,
-                "ROLLBACK_IN_PROGRESS": ComputeFleetStatus.STOPPING,
-                "ROLLBACK_FAILED": ComputeFleetStatus.STOPPED,
-                "ROLLBACK_COMPLETE": ComputeFleetStatus.STOPPED,
-                "DELETE_FAILED": ComputeFleetStatus.STOPPING,
-            }
-            return stack_status_to_fleet_status.get(self.status, status)
+        status, _ = self.compute_fleet_status_with_last_updated_time
         return status
 
+    @property
+    def compute_fleet_status_with_last_updated_time(self):
+        """Status of the cluster compute fleet and the last compute fleet status updated time."""
+        if self.stack.scheduler == "slurm":
+            compute_fleet_status_manager = ComputeFleetStatusManager(self.name)
+            status, last_updated_time = compute_fleet_status_manager.get_status_with_last_updated_time()
+            if status == ComputeFleetStatus.UNKNOWN:
```

Comment:
I'm wondering if it makes more sense to chance this logic and just return UNKNOWN as status when the dynamodb entry is not available (and we can do the same for batch Compute environment). In this way we return an empty lastUpdatedTime only when the status is UNKNOWN

Reply:
To make it `UNKNOWN`, you want to get rid of the mapping from Stack status to Compute fleet status below?

Reply:
Let's talk about this if my comment does not make sense

Reply:
Done